sql: TestTransactionDeadline failed

sql.TestTransactionDeadline [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/19091479?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/19091479?buildTab=artifacts#/) on master @ [6f4f374ac2869fcd7084864411e9bb217c14d4c6](https://github.com/cockroachdb/cockroach/commits/6f4f374ac2869fcd7084864411e9bb217c14d4c6):


```
=== RUN   TestTransactionDeadline
    test_log_scope.go:165: test logs captured to: /artifacts/tmp/_tmp/4e9b6c662b8a17b1b0bf70dc9d006949/logTestTransactionDeadline2248296157
    test_log_scope.go:76: use -show-logs to present logs inline
    conn_executor_test.go:1428: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/4e9b6c662b8a17b1b0bf70dc9d006949/logTestTransactionDeadline2248296157
--- FAIL: TestTransactionDeadline (1.11s)
=== RUN   TestTransactionDeadline/session_expiry_overrides_lease_deadline
    conn_executor_test.go:1400: 
        	Error Trace:	pkg/sql/conn_executor_test.go:1400
        	            				pkg/sql/conn_executor_test.go:1299
        	            				pkg/sql/conn_executor_test.go:1400
        	Error:      	Not equal: 
        	            	expected: hlc.Timestamp{WallTime:1741881122901401554, Logical:0}
        	            	actual  : hlc.Timestamp{WallTime:1741880922498336395, Logical:0}
        	            	
        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -1,3 +1,3 @@
        	            	 (hlc.Timestamp) {
        	            	- WallTime: (int64) 1741881122901401554,
        	            	+ WallTime: (int64) 1741880922498336395,
        	            	  Logical: (int32) 0
        	Test:       	TestTransactionDeadline/session_expiry_overrides_lease_deadline
--- FAIL: TestTransactionDeadline/session_expiry_overrides_lease_deadline (0.03s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestTransactionDeadline.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-48541

sql.TestTransactionDeadline [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/19164162?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/19164162?buildTab=artifacts#/) on master @ [65d78006cac938bc8a6cd599847cab190a6952ae](https://github.com/cockroachdb/cockroach/commits/65d78006cac938bc8a6cd599847cab190a6952ae):


```
=== RUN   TestTransactionDeadline
    test_log_scope.go:165: test logs captured to: /artifacts/tmp/_tmp/4e9b6c662b8a17b1b0bf70dc9d006949/logTestTransactionDeadline4040617494
    test_log_scope.go:76: use -show-logs to present logs inline
    conn_executor_test.go:1428: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/4e9b6c662b8a17b1b0bf70dc9d006949/logTestTransactionDeadline4040617494
--- FAIL: TestTransactionDeadline (0.91s)
=== RUN   TestTransactionDeadline/session_expiry_overrides_lease_deadline
    conn_executor_test.go:1400: 
        	Error Trace:	pkg/sql/conn_executor_test.go:1400
        	            				pkg/sql/conn_executor_test.go:1299
        	            				pkg/sql/conn_executor_test.go:1400
        	Error:      	Not equal: 
        	            	expected: hlc.Timestamp{WallTime:1742507531843587820, Logical:0}
        	            	actual  : hlc.Timestamp{WallTime:1742507331548688070, Logical:0}
        	            	
        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -1,3 +1,3 @@
        	            	 (hlc.Timestamp) {
        	            	- WallTime: (int64) 1742507531843587820,
        	            	+ WallTime: (int64) 1742507331548688070,
        	            	  Logical: (int32) 0
        	Test:       	TestTransactionDeadline/session_expiry_overrides_lease_deadline
--- FAIL: TestTransactionDeadline/session_expiry_overrides_lease_deadline (0.01s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestTransactionDeadline.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

--- a/pkg/sql/BUILD.bazel
+++ b/pkg/sql/BUILD.bazel
@@ -891,6 +891,7 @@ go_test(
         "//pkg/sql/sqlerrors",
         "//pkg/sql/sqlinstance",
         "//pkg/sql/sqlliveness",
+        "//pkg/sql/sqlliveness/slbase",
         "//pkg/sql/sqlliveness/sqllivenesstestutils",
         "//pkg/sql/sqlstats",
         "//pkg/sql/sqlstats/persistedsqlstats",

--- a/pkg/sql/conn_executor_test.go
+++ b/pkg/sql/conn_executor_test.go
@@ -40,6 +40,7 @@ import (
 	"github.com/cockroachdb/cockroach/pkg/sql/sem/tree"
 	"github.com/cockroachdb/cockroach/pkg/sql/sessiondata"
 	"github.com/cockroachdb/cockroach/pkg/sql/sqlliveness"
+	"github.com/cockroachdb/cockroach/pkg/sql/sqlliveness/slbase"
 	"github.com/cockroachdb/cockroach/pkg/sql/sqlliveness/sqllivenesstestutils"
 	"github.com/cockroachdb/cockroach/pkg/testutils"
 	"github.com/cockroachdb/cockroach/pkg/testutils/pgtest"
@@ -1354,6 +1355,14 @@ func TestTransactionDeadline(t *testing.T) {
 			SessionOverride: sessionOverrideKnob,
 		},
 	}
+	// Previously, this test could flake if the actual session
+	// had a shorter TTL remaining then the fake session. This
+	// would cause the txn to pick a shorter deadline then our fake
+	// session. So, intentionally change the actual session TTL
+	// to a really long time.
+	st := cluster.MakeClusterSettings()
+	slbase.DefaultTTL.Override(ctx, &st.SV, time.Minute*5)
+
 	s := serverutils.StartServerOnly(t, base.TestServerArgs{
 		DefaultTestTenant: base.TestControlsTenantsExplicitly,
 		Knobs:             knobs,
@@ -1366,6 +1375,7 @@ func TestTransactionDeadline(t *testing.T) {
 		base.TestTenantArgs{
 			TenantID:     serverutils.TestTenantID(),
 			TestingKnobs: knobs,
+			Settings:     st,
 		})
 	tdb := sqlutils.MakeSQLRunner(sqlConn)
 	// Set up a dummy database and table in the tenant to write to.

